framework/src/base/AuxiliarySystem.C

```diff
@@ -396,14 +405,17 @@ AuxiliarySystem::computeNodalVars(ExecFlagType type)
       solution().close();
       _sys.update();
     }
+    PARALLEL_CATCH;
+    Moose::perf_log.pop(compute_aux_tag, "Execution");
   }
-  PARALLEL_CATCH;
-  Moose::perf_log.pop("update_aux_vars_nodal()", "Execution");
 
-  // Boundary Nodal AuxKernels
-  Moose::perf_log.push("update_aux_vars_nodal_bcs()", "Execution");
-  PARALLEL_TRY {
-    if (nodal.hasActiveBoundaryObjects())
+  if (nodal.hasActiveBoundaryObjects())
```

Comment:
Are we sure that all processors always return the same result for `if (nodal.hasActiveBoundaryObjects())` et al?  If not, moving the `PARALLEL_CATCH` inside the if-statement will cause the code to hang.